FWCore/src/components/PileupMerge.cpp

```diff
@@ -0,0 +1,117 @@
+  StatusCode sc = GaudiAlgorithm::finalize();
+  return sc;
+}
+
+
+StatusCode PileupMerge::execute() {
+  unsigned nEvents = m_reader.getEntries();
+
+
+  const fcc::TrackHitCollection* trackHitCollection;
+  fcc::TrackHitCollection* hitCollOut = new fcc::TrackHitCollection();
+
+  const fcc::PositionedTrackHitCollection* posTrackHitCollection;
+  fcc::PositionedTrackHitCollection* posHitCollOut = new fcc::PositionedTrackHitCollection();
+
+  const fcc::MCParticleCollection* mcParticleCollection;
+  fcc::MCParticleCollection* mcPartCollOut = new fcc::MCParticleCollection();
+
+  const fcc::GenVertexCollection* genVertexCollection;
+  fcc::GenVertexCollection* genVertCollOut = new fcc::GenVertexCollection();
+
+  const fcc::CaloHitCollection* caloHitCollection;
+  fcc::CaloHitCollection* caloHitCollOut = new fcc::CaloHitCollection();
+
+  const fcc::PositionedCaloHitCollection* posCaloHitCollection;
+  fcc::PositionedCaloHitCollection* posCaloHitCollOut = new fcc::PositionedCaloHitCollection();
+
+  const unsigned int numPileUp = m_pileUpTool->numberOfPileUp();
+  for(unsigned iev = 0; iev < numPileup; ++iev) {
+
+    if ( m_randomizePileup == true ) {
+      if (m_flatDist() < 0.3) { // skip one in three events
+        ++currentEvent;
+      }
+    }
+    currentEvent = (currentEvent + 1) % nEvents; // start over from beginning if necessary
+    m_reader.goToEvent(currentEvent);
+  }
+  
+    bool mcParticleCollectionPresent = m_store.get(m_pileupGenParticlesBranchName, mcParticleCollection);
+    if (mcParticleCollectionPresent) {
+      for (auto elem: (*mcParticleCollection)) {
+        mcPartCollOut->push_back(elem.clone());
+      }
+    }
+
+    bool genVertexCollectionPresent = m_store.get(m_pileupGenVerticesBranchName, genVertexCollection);
+    if (genVertexCollectionPresent) {
+      for (auto elem: (*genVertexCollection)) {
+        genVertCollOut->push_back(elem.clone());
+      }
+    }
+
+    bool trackHitsPresent = m_store.get(m_pileupTrackHitsBranchName, trackHitCollection);
+    if (trackHitsPresent) {
+      for (auto elem: (*trackHitCollection)) {
+        hitCollOut->push_back(elem.clone());
+      }
+    }
+
+    bool posTrackHitsPresent = m_store.get(m_pileupPositionedTrackHitsBranchName, posTrackHitCollection);
+    if (posTrackHitsPresent) {
+      for (auto elem: (*posTrackHitCollection)) {
+        posHitCollOut->push_back(elem.clone());
+      }
+    }
+
+    bool caloHitsPresent = m_store.get(m_pileupCaloHitsBranchName, caloHitCollection);
+    if (caloHitsPresent) {
+      for (auto elem: (*caloHitCollection)) {
+        caloHitCollOut->push_back(elem.clone());
+      }
+    }
+
+    bool posCaloHitsPresent = m_store.get(m_pileupPositionedCaloHitsBranchName, posCaloHitCollection);
+    if (posCaloHitsPresent) {
+      for (auto elem: (*posCaloHitCollection)) {
+        posCaloHitCollOut->push_back(elem.clone());
+      }
+    }
+
+
+    m_store.clear();
+
+  m_trackHitsOut.put(hitCollOut);
```

Comment:
We can do this for all the collections, right? Or add a switch so that an user can say what collections he is interested in.